x-pack/plugins/ml/public/application/components/severity_control/severity_control.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiRange,
+  EuiRangeProps,
+} from '@elastic/eui';
+import { ANOMALY_THRESHOLD } from '../../../../common';
+import './styles.scss';
+
+export interface SeveritySelectorProps {
+  value: number | undefined;
+  onChange: (value: number) => void;
+}
+
+const MAX_ANOMALY_SCORE = 100;
+
+export const SeverityControl: FC<SeveritySelectorProps> = React.memo(({ value, onChange }) => {
+  const levels: EuiRangeProps['levels'] = [
+    {
+      min: ANOMALY_THRESHOLD.LOW,
+      max: ANOMALY_THRESHOLD.MINOR - 1,
+      color: 'success',
+    },
+    {
+      min: ANOMALY_THRESHOLD.MINOR,
+      max: ANOMALY_THRESHOLD.MAJOR - 1,
+      color: 'primary',
+    },
+    {
+      min: ANOMALY_THRESHOLD.MAJOR,
+      max: ANOMALY_THRESHOLD.CRITICAL,
+      color: 'warning',
+    },
+    {
+      min: ANOMALY_THRESHOLD.CRITICAL,
+      max: MAX_ANOMALY_SCORE,
+      color: 'danger',
+    },
+  ];
+
+  const label = i18n.translate('xpack.ml.severitySelector.formControlLabel', {
+    defaultMessage: 'Severity',
+  });
+
+  const resultValue = value ?? ANOMALY_THRESHOLD.LOW;
+
+  const onChangeCallback = (
+    e: React.ChangeEvent<HTMLInputElement> | React.MouseEvent<HTMLButtonElement>
+  ) => {
+    // @ts-ignore Property 'value' does not exist on type 'EventTarget' | (EventTarget & HTMLInputElement)
+    onChange(Number(e.target.value));
+  };
+
+  return (
+    <EuiFormRow fullWidth>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiFieldNumber
+            id="severityControl"
+            style={{ width: '70px' }}
+            compressed
+            prepend={label}
+            value={resultValue}
+            onChange={onChangeCallback}
+            min={ANOMALY_THRESHOLD.LOW}
+            max={MAX_ANOMALY_SCORE}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={true}>
+          <EuiRange
+            className={'mlSeverityControl'}
+            fullWidth
+            min={ANOMALY_THRESHOLD.LOW}
+            max={MAX_ANOMALY_SCORE}
+            value={resultValue}
+            onChange={onChangeCallback}
+            aria-label={i18n.translate('xpack.ml.severitySelector.formControlAriaLabel', {
+              defaultMessage: 'Select severity threshold',
+            })}
+            showTicks
+            tickInterval={10}
```

Comment:
For me (Chrome and FF, Windows) there seems to be a nasty usability issue with `EuiRange` if you specify `tickInterval` rather than specific `ticks` where the `button` control rendered for the tick label can become so wide that it prevents the user interacting with other controls on the page (here shown with a tickInterval of `50`):

![image](https://user-images.githubusercontent.com/7405507/119971748-47ce1b00-bfa9-11eb-9ab8-9e6aee35dd25.png)

For me this is fixed by replacing `tickInterval` with fixed ticks, such as (I quite like using ticks at interval of `25` as these align with the severity ranges):

```
            ticks={[
              { label: '0', value: 0 },
              { label: '25', value: 25 },
              { label: '50', value: 50 },
              { label: '75', value: 75 },
              { label: '100', value: 100 },
            ]}
```
which means the `button` controls do not get assigned that problematic `width`:
![image](https://user-images.githubusercontent.com/7405507/119972055-a85d5800-bfa9-11eb-971e-479a2a7ae881.png)

Reply:
Fixed in https://github.com/elastic/kibana/pull/99489/commits/218a74c9bd6cc501e801dc9e679788bfe05c8402